In [18]:
import re
#import sketch
import numpy as np 
import pandas as pd 
#import pandas_alive
import matplotlib.pyplot as plt
import seaborn as sns

Load files

In [19]:
facts = pd.read_excel("data/homicidios.xlsx", sheet_name="HECHOS")
victims = pd.read_excel("data/homicidios.xlsx", sheet_name="VICTIMAS")

Treatment of data from the dataset 'facts'

In [20]:
facts

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA  HH  \
0    2016-0001           1 2016-01-01  2016   1   1  04:00:00   4   
1    2016-0002           1 2016-01-02  2016   1   2  01:15:00   1   
2    2016-0003           1 2016-01-03  2016   1   3  07:00:00   7   
3    2016-0004           1 2016-01-10  2016   1  10  00:00:00   0   
4    2016-0005           1 2016-01-21  2016   1  21  05:20:00   5   
..         ...         ...        ...   ...  ..  ..       ...  ..   
691  2021-0093           1 2021-12-13  2021  12  13  17:10:00  17   
692  2021-0094           1 2021-12-20  2021  12  20  01:10:00   1   
693  2021-0095           1 2021-12-30  2021  12  30  00:43:00   0   
694  2021-0096           1 2021-12-15  2021  12  15  10:30:00  10   
695  2021-0097           1 2021-11-18  2021  11  18  06:10:00   6   

                               LUGAR_DEL_HECHO TIPO_DE_CALLE  ...  Altura  \
0    AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...     NaN   
1            AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...     NaN   
2                           AV ENTRE RIOS 2034       AVENIDA  ...  2034.0   
3        AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  ...     NaN   
4    AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  ...     NaN   
..                                         ...           ...  ...     ...   
691                          AV. RIESTRA Y MOM       AVENIDA  ...     NaN   
692                    AU DELLEPIANE Y LACARRA     AUTOPISTA  ...     NaN   
693                        AV. GAONA Y TERRADA       AVENIDA  ...     NaN   
694                         AV. EVA PERON 4071       AVENIDA  ...  4071.0   
695                    PADRE CARLOS MUJICA 709         CALLE  ...   709.0   

                                   Cruce  \
0    FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                    DE LOS CORRALES AV.   
2                                    NaN   
3               VILLEGAS, CONRADO, GRAL.   
4                SAENZ PE?A, LUIS, PRES.   
..                                   ...   
691                                  MOM   
692                          LACARRA AV.   
693                              TERRADA   
694                                  NaN   
695                                  NaN   

                                 Dirección Normalizada COMUNA  \
0    PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1                 PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                  ENTRE RIOS AV. 2034      1   
3            LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.      8   
4              SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.      1   
..                                                 ...    ...   
691                                  RIESTRA AV. y MOM      7   
692         DELLEPIANE, LUIS, TTE. GRAL. y LACARRA AV.      9   
693                                GAONA AV. y TERRADA     11   
694                                PERON, EVA AV. 4071      9   
695                            PADRE CARLOS MUGICA 709      1   

                                   XY (CABA)         pos x         pos y  \
0      Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1      Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2     Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3      Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4    Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   
..                                       ...           ...           ...   
691   Point (102728.60090138 98186.24929177)  -58.43353773  -34.64561636   
692    Point (99624.29795829 97569.69801131)  -58.46739825  -34.65117757   
693   Point (99116.45492358 101045.23284826)  -58.47293407  -34.61984745   
694    Point (99324.54463985 97676.26932409)  -58.47066794  -34.65021673   
695  Point (107664.16647795 104708.63962087)  -58.37976155  -34.58679619   

      PARTICIPANTES    VICTIMA    A

In [21]:
def data_summ(df):
    '''
    This function provides detailed information about the dtype and null values present in a dataframe
    '''

    info_dict = {"Column": [], "Data_type": [], "No_miss_Qty": [], "%Missing": [], "Missing_Qty": []}
    

    for column in df.columns:
        
        info_dict["Column"].append(column)
        info_dict["Data_type"].append(df[column].apply(type).unique())
        info_dict["No_miss_Qty"].append(df[column].count())
        info_dict["%Missing"].append(round(df[column].isnull().sum() * 100 / len(df), 2))
        info_dict['Missing_Qty'].append(df[column].isnull().sum())

    df_info = pd.DataFrame(info_dict)
    
    print("\nTotal rows: ", len(df))
    print("\nTotal full null rows: ", df.isna().all(axis=1).sum())
    print("\nTotal duplicated rows: ", df.duplicated().sum())
    
    return df_info

In [22]:
data_summ(facts)


Total rows:  696

Total full null rows:  0

Total duplicated rows:  0


Column                                          Data_type  \
0                      ID                                    [<class 'str'>]   
1              N_VICTIMAS                                    [<class 'int'>]   
2                   FECHA  [<class 'pandas._libs.tslibs.timestamps.Timest...   
3                    AAAA                                    [<class 'int'>]   
4                      MM                                    [<class 'int'>]   
5                      DD                                    [<class 'int'>]   
6                    HORA  [<class 'datetime.time'>, <class 'str'>, <clas...   
7                      HH                     [<class 'int'>, <class 'str'>]   
8         LUGAR_DEL_HECHO                                    [<class 'str'>]   
9           TIPO_DE_CALLE                                    [<class 'str'>]   
10                  Calle                   [<class 'str'>, <class 'float'>]   
11                 Altura                                  [<class 'float'>]   
12                  Cruce                   [<class 'str'>, <class 'float'>]   
13  Dirección Normalizada                   [<class 'str'>, <class 'float'>]   
14                 COMUNA                                    [<class 'int'>]   
15              XY (CABA)                                    [<class 'str'>]   
16                  pos x                                    [<class 'str'>]   
17                  pos y                                    [<class 'str'>]   
18          PARTICIPANTES                                    [<class 'str'>]   
19                VICTIMA                                    [<class 'str'>]   
20                ACUSADO                                    [<class 'str'>]   

    No_miss_Qty  %Missing  Missing_Qty  
0           696      0.00            0  
1           696      0.00            0  
2           696      0.00            0  
3           696      0.00            0  
4           696      0.00            0  
5           696      0.00            0  
6           696      0.00            0  
7           696      0.00            0  
8           696      0.00            0  
9           696      0.00            0  
10          695      0.14            1  
11          129     81.47          567  
12          525     24.57          171  
13          688      1.15            8  
14          696      0.00            0  
15          696      0.00            0  
16          696      0.00            0  
17          696      0.00            0  
18          696      0.00            0  
19          696      0.00            0  
20          696      0.00            0

## Columns with null data are analyzed

- Column 'Calle' has 1 null data
- Column 'Altura' has 567 null data
- Column 'Cruce' has 171 null data
- Column 'Dirección Normalizada' has 8 null data

In [23]:
facts[facts["Calle"].isnull()][["Dirección Normalizada", "Cruce","Altura", "Calle"]]

Dirección Normalizada Cruce  Altura Calle
119                   NaN   NaN     NaN   NaN

In [24]:
facts[facts["Dirección Normalizada"].isnull()][["Dirección Normalizada", "Cruce","Altura", "Calle"]]

Dirección Normalizada Cruce  Altura                                 Calle
38                    NaN   NaN     NaN                 LUGONES, LEOPOLDO AV.
106                   NaN   NaN     NaN     AUTOPISTA BUENOS AIRES - LA PLATA
119                   NaN   NaN     NaN                                   NaN
180                   NaN   NaN     NaN               AUTOPISTA PERITO MORENO
181                   NaN   NaN     NaN  AUTOPISTA DELLEPIANE LUIS TTE. GRAL.
313                   NaN   NaN     NaN                 LUGONES, LEOPOLDO AV.
546                   NaN   NaN     NaN                 LUGONES, LEOPOLDO AV.
621                   NaN   NaN     NaN     AUTOPISTA BUENOS AIRES - LA PLATA


Analyzing the data from the 'Altura' column, we proceed to eliminate it due to the large number of missing items and there being no way to fill in the columns; Investigating the data in the 'Altura' column, we have seen that it refers to the numbering of the street where the accident occurs. The dataset shows us that the majority of accidents occur at intersections or corners

In [25]:
# Drop column "Altura"
facts= facts.drop("Altura", axis=1)

In [27]:
facts.head()

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA HH  \
0  2016-0001           1 2016-01-01  2016   1   1  04:00:00  4   
1  2016-0002           1 2016-01-02  2016   1   2  01:15:00  1   
2  2016-0003           1 2016-01-03  2016   1   3  07:00:00  7   
3  2016-0004           1 2016-01-10  2016   1  10  00:00:00  0   
4  2016-0005           1 2016-01-21  2016   1  21  05:20:00  5   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE             Calle  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  PIEDRA BUENA AV.   
1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ    PAZ, GRAL. AV.   
2                         AV ENTRE RIOS 2034       AVENIDA    ENTRE RIOS AV.   
3      AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA    LARRAZABAL AV.   
4  AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA      SAN JUAN AV.   

                                 Cruce  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                  DE LOS CORRALES AV.   
2                                  NaN   
3             VILLEGAS, CONRADO, GRAL.   
4              SAENZ PE?A, LUIS, PRES.   

                               Dirección Normalizada  COMUNA  \
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...       8   
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.       9   
2                                ENTRE RIOS AV. 2034       1   
3          LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.       8   
4            SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.       1   

                                 XY (CABA)         pos x         pos y  \
0    Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1    Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2   Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3    Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4  Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   

    PARTICIPANTES VICTIMA    ACUSADO  
0       MOTO-AUTO    MOTO       AUTO  
1  AUTO-PASAJEROS    AUTO  PASAJEROS  
2       MOTO-AUTO    MOTO       AUTO  
3         MOTO-SD    MOTO         SD  
4  MOTO-PASAJEROS    MOTO  PASAJEROS